### DISCUSSION TITLE
Identity users and roles in host and tenant database...

### DISCUSSION THREAD

**Discussion by dledob** [abpframework/abp#20299] [Q&A] [answered]
Currently I have a one context for the host database and another for the tenant databases, in both contexts I use the identity module, according to the documentation "Some modules, like the [Identity](https://abp.io/docs/latest/modules/identity) module, is both used in host and tenant sides".  To create roles and users I use the services IdentityUserManager and IdentityRoleManager as shown in the image below
![image](https://github.com/user-attachments/assets/6bff61c8-fd63-4338-a8eb-cff729e4c8ac)
The first time the code is executed, the user and role are added correctly only in the tenant database, but when I run it again, the users and roles are duplicated because FindByEmailAsync and FindByNameAsync do not find them, the IdentityUserManager and IdentityRoleManager It seems they are searching in the host database, they persist Ok but they do not query the correct database. 
It is possible to use these services to correctly query the users and roles in the tenant database? thank you so much.

**Comment by maliming** [comment, marked as answer]
hi

Please refer to [SetMultiTenancySide](https://abp.io/docs/latest/framework/data/entity-framework-core/migrations#separating-host-tenant-database-schemas)

**Comment by dledob** [comment]
thank you very much [maliming], also it had to be disabled the tenant filter in module config
![image](https://github.com/user-attachments/assets/17f14a33-2e5c-4e8a-8907-37e4419bf369)
